### PR TITLE
fix for registering nested message types

### PR DESCRIPTION
**Pull request by thiagodpf**
@mstoykov Related to [this](https://github.com/grafana/k6/issues/3160) issue.

### CODE REVIEW

**Review by olegbespalov — COMMENTED** [review]
Hi @thiagodpf !

First of all, I'd like to thank you for raising the issue and working on a fix :bow: Great job!

Regarding the PR, it looks good :+1:  

However, the one possible improvement could be implementing the stack here instead of using the github.com/golang-collections/collections package (the go implementation is pretty straightforward). The rationale is to reduce the number of libraries we import, improving maintainability. 

How does that sound?

**Comment by thiagodpf** [comment]
Hi @olegbespalov !!
I think it's great! I will do this!
Do you see any problems if I bring in the `stack` implementation from the `golang-collections` library as-is?
Another point is, is there a file/directory where the `stack` is better placed or should I include it in the `client.go` file itself?

File: grpc/client.go [hunk 1]
```diff
@@ -282,17 +284,30 @@ func (c *Client) convertToMethodInfo(fdset *descriptorpb.FileDescriptorSet) ([]M
 				appendMethodInfo(fd, sd, md)
 			}
 		}
+
+		s := stack.New()
```
Comment (mstoykov) [inline grpc/client.go:288]: I would just make this a slice and will `append` on `push` and on pop I will `message = s[len(s)-1]; s = s[:len(s)-1]` 

This is all we need in practice. This also will help with the type safety

Comment (thiagodpf) [reply grpc/client.go:288]: Perfect! 🆙
Better than that, impossible! 🚀
Thank you for your help!

**Review by olegbespalov — APPROVED** [review]
Looks good to me! :rocket: 

Thank you for your contribution! :bow:

**Review by mstoykov — APPROVED** [review]
Thanks for the work on this :bow:

**Comment by olegbespalov** [comment]
@thiagodpf just a tiny update on the next steps :+1: 

We're going to pack some more changes into the xk6-grpc "release" and once it is done, bump the new version in the primary k6's repo. Once that is done, the k6's experimental module will get the fixes. 

In parallel, we're going to backport the changes to [the legacy `k6/net/grpc` module](https://github.com/grafana/k6/blob/master/js/modules/k6/grpc/client.go). For sure if you want you could do the last part also by yourself, but according to our freshly posted [contribution guidelines](https://github.com/grafana/xk6-grpc#contributing) we don't expect this.

Thanks again! :rocket:

**Comment by thiagodpf** [comment]
@olegbespalov thanks for accepting my PR!
So in a few hours I will backport the PR into the legacy module.
Thanks for all the help!

**Comment by thiagodpf** [comment]
Done! :v::smiley:
https://github.com/grafana/k6/pull/3178

Again, thanks for the support and help!

**Pull request merged**